### PR TITLE
deterministic sac on inference

### PR DESCRIPTION
**Pull request by AlexGrinch**
SAC should produce deterministic actions on inference. Also it seems that some of set_global_seeds calls were superfluous.

### CODE REVIEW

File: catalyst/rl/offpolicy/sampler.py [hunk 1]
```diff
@@ -276,7 +276,6 @@ def run(self):
         seed = random.randrange(SEED_RANGE) \
             if self.seeds is None \
             else random.choice(self.seeds)
-        set_global_seeds(seed)
```
Comment (Scitator) [inline catalyst/rl/offpolicy/sampler.py]: why have you removed seed specification for new episode?

File: catalyst/rl/agents/layers.py [hunk 2]
```diff
@@ -177,9 +177,9 @@ def _forward_lama(self, observation):
         x = self.head_net(x)
         return x
 
-    def forward(self, *args, with_log_pi=False, **kwargs):
+    def forward(self, *args, with_log_pi=False, deterministic=False, **kwargs):
```
Comment (Scitator) [inline catalyst/rl/agents/layers.py:180]: looks like we need to think about better forward API for agents... in future :)

File: catalyst/rl/offpolicy/sampler.py [hunk 3]
```diff
@@ -415,7 +414,6 @@ def run(self):
                     seed = random.randrange(SEED_RANGE)
             else:
                 seed = random.choice(self.seeds)
-            set_global_seeds(seed)
```
Comment (Scitator) [inline catalyst/rl/offpolicy/sampler.py]: and here?

Comment (AlexGrinch) [reply catalyst/rl/offpolicy/sampler.py]: Returned for further investigation

File: catalyst/rl/agents/layers.py (hunk 2 above)
Comment (AlexGrinch) [reply catalyst/rl/agents/layers.py:180]: definitely

**Pull request merged**